coconut/src/proof.rs

```diff
@@ -42,46 +42,50 @@ impl RequestCoinsProof {
         output_attributes: &[OutputAttribute],
         randomness: &Randomness,
     ) -> Self {
-        assert!(parameters.max_attributes() >= 2);
-        assert!(public_key.max_attributes() >= 2);
+        assert!(parameters.max_attributes() >= 3);
+        assert!(public_key.max_attributes() >= 3);
         assert!(parameters.max_attributes() >= output_attributes.len());
 
         // Compute the witnesses.
         let input_attributes_witnesses: Vec<_> = input_attributes
             .iter()
             .map(|_| InputAttribute {
                 value: Scalar::random(&mut rng),
-                id: Scalar::random(&mut rng),
+                seed: Scalar::random(&mut rng),
+                id: Scalar::zero(), // Never used, the input id is revealed.
```

Comment:
It is because we re-use the same data struct as for the Input Attributes. Witnesses take the same 'shape' as attributes but are random. Perhaps a cleaner way is to separate "hidden attributes" and "public attributes" and have two data structures

Reply:
> Perhaps a cleaner way is to separate "hidden attributes" and "public attributes" and have two data structures

Ok, that's what I thought. Up to you